DBI.xs

```diff
@@ -4066,7 +4073,7 @@ XS(XS_DBI_dispatch)
         }
 
         hook_svp = NULL;
-        if (    SvTRUE(err_sv)
+        if (   (SvTRUE(err_sv) || (is_warning && DBIc_has(imp_xxh, DBIcf_RaiseWarn)))
```

Comment:
With this change HandleError will be called for warnings but only if RaiseWarn is true.
That seems surprising, and also not mentioned in the docs.
You could revert this hunk and remove the mention of !hook_svp in the hunk below, or update the docs and provide a good rationale for the behaviour.
I think I'd prefer the former.

Reply:
Currently `HandleError` is **not** called for warnings even when `PrintWarn` is set to true.

My change adds a new `RaiseWarn` attribute which is by default false. So default behavior of `HandleError` is not changed, it is not called for warnings.

When `RaiseWarn` is set to true value, then all warnings are `fatal` similar to errors and I think it make sense to send them into `HandleError` callback. So users can handle in `HandleError` anything which is going to throw an exception from DBI. I think this is better behavior and expected behavior. If you agree I update information about it in documentation.

Reply:
I updated documentation for `HandleError` and rebased this patch on top of master branch.

Reply:
@timbunce Is it now better?